Documentation/ceph-cluster-crd.md

```diff
@@ -1489,3 +1490,31 @@ data:
 
 Note: if you are using self-signed certificates (not known/approved by a proper CA) you must pass `VAULT_SKIP_VERIFY: true`.
 Communications will remain encrypted but the validity of the certificate will not be verified.
+
+##### Ceph CSI configuration
+
+Ceph-CSI is capable of encrypting PVs requests.
+Rook can configure Ceph-CSI to turn on encryption with Key Management System support.
+For this, you need to add new parameter in your CephCluster CR spec:
+
+```yaml
+security:
+  kms:
+  ...
+  ...
+  csiConnectionDetailsCMName: "rook-ceph-csi-vault"
+```
+
+The config map will look like:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rook-ceph-csi-vault
+  namespace: rook-ceph
+data:
+  csiConfig: TODO
```

Comment:
@Madhu-1 can you help here? Thanks!

Reply:
@nixpanic can you paste an example for this please?

Reply:
At the moment, I think that ceph/ceph-csi#1743 contains a complete configuration. This would be placed in a ConfigMap like [kms-config.yaml](https://github.com/ceph/ceph-csi/blob/master/examples/kms/vault/kms-config.yaml).

- Rook would not fill the `tenants` and leave it out
- The name `vault-with-tokens` should probably match the service name that is given to the Vault connection, or default to something Rook specific (e.g. `rook-vault-tokens`)
- Note that the ConfigMap may contain other KMS configurations too (it is a list), the `kms-config.yaml` from the example directory will get extended with a 2nd entry too. Rook should be careful not to overwrite/remove other configurations from the file.

---

### Example of the KMS configuration file for Vault Tokens

The configuration is available in the Ceph-CSI containers as
`/etc/ceph-csi-encryption-kms-config/config.json`:

```json
{
    "vault-with-tokens": {
        "encryptionKMSType": "vaulttokens",
        "vaultAddress": "http://vault.default.svc.cluster.local:8200",
        "vaultCAFromSecret": "vault-ca",
        "vaultCAVerify": "false",
        "configName": "ceph-csi-kms-config",
        "tokenName": "ceph-csi-kms-token",
        "tenants": {
            "my-app": {
                "vaultAddress": "https://vault.example.com",
                "vaultCAVerify": "true"
            },
            "an-other-app": {
                "tokenName": "storage-encryption-token"
            }
        }
    }
}
```

- `tenants`: list of Tenants (Kubernetes Namespaces) with their Vault Token and
  connection configuration (Administrator managed, optional)
- `configName`: the name of the Kubernetes ConfigMap that contains the Vault
  connection configuration (optional, can be Tenant managed)
- `tokenName`: the name of the Kubernetes Secret that contains the Vault
  Token (optional, can be Tenant managed)

Reply:
@nixpanic @Madhu-1 PTAL, did I get it right? @Madhu-1 can still change that in his followup PR. Thanks